core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/rds/RdsUpgradeService.java

```diff
@@ -96,9 +123,25 @@ private RdsUpgradeV4Response validateAndTrigger(StackView stack, TargetMajorVers
             LOGGER.warn("Stack {} is not available for RDS upgrade", stack.getName());
             throw new BadRequestException(getMessage(CLUSTER_RDS_UPGRADE_NOT_AVAILABLE));
         }
+        validateRequestTime(stack, accountId);
+    }
 
-        LOGGER.info("External database for stack {} will be upgraded to version {}", stack.getName(), targetMajorVersion.getMajorVersion());
-        return triggerRdsUpgradeFlow(stack, targetMajorVersion);
+    private void validateRequestTime(StackView stack, String accountId) {
+        if (entitlementService.isPostgresUpgradeAttachedDatahubsCheckEnabled(accountId) && stack.isDatalake()) {
+            Set<StackListItem> datahubs = stackService.getByWorkspaceId(restRequestThreadLocalService.getRequestedWorkspaceId(), stack.getEnvironmentCrn(),
+                    List.of(StackType.WORKLOAD));
+            String notStoppedAttachedClusters = datahubs.stream()
+                    .filter(datahub -> !Status.getAllowedDataHubStatesForSdxUpgrade().contains(datahub.getStackStatus())
+                            || !Status.getAllowedDataHubStatesForSdxUpgrade().contains(datahub.getClusterStatus()))
+                    .map(StackListItem::getName)
+                    .collect(Collectors.joining(","));
+            if (StringUtils.isNotEmpty(notStoppedAttachedClusters)) {
+                String msg = String.format("There are attached Data Hub clusters in incorrect state: %s. "
+                        + "Please stop those to be able to perform the upgrade.", notStoppedAttachedClusters);
```

Comment:
I would add: "... to perform the database server upgrade."
Or rds upgrade.
We should also agree on how this is called:
- rds upgrade
- database upgrade
- database server upgrade

This is irrespective of the current PR, I think error message use all kind of terminilogy.